src/explore-education-statistics-admin/src/pages/release/datablocks/components/chart/__tests__/ChartBoundaryLevelsConfiguration.test.tsx

```diff
@@ -63,75 +90,180 @@ describe('ChartBoundaryLevelsConfiguration', () => {
     );
   }
 
-  test('renders correctly without initial values', () => {
+  test('renders without data sets table', () => {
+    render(
+      <ChartBoundaryLevelsConfiguration
+        map={{ dataSetConfigs: [] }}
+        meta={testMeta}
+        options={testDefaultChartOptions}
+        onChange={noop}
+        onSubmit={noop}
+        hasDataSetBoundaryLevels
+      />,
+    );
+    expect(screen.getByLabelText('Default boundary level')).toBeInTheDocument();
+    expect(
+      screen.queryByText('Set boundary levels per data set'),
+    ).not.toBeInTheDocument();
+    expect(screen.queryByRole('table')).not.toBeInTheDocument();
+  });
+
+  test('renders data sets without initial boundary levels', () => {
     render(
       <ChartBoundaryLevelsConfiguration
+        map={testDefaultMap}
         meta={testMeta}
         options={testDefaultChartOptions}
         onChange={noop}
         onSubmit={noop}
+        hasDataSetBoundaryLevels
       />,
     );
 
-    expect(screen.getByLabelText('Boundary level')).not.toHaveValue();
-    const boundaryLevels = within(
-      screen.getByLabelText('Boundary level'),
-    ).getAllByRole('option');
-
-    expect(boundaryLevels).toHaveLength(4);
-    expect(boundaryLevels[0]).toHaveTextContent('Please select');
-    expect(boundaryLevels[0]).toHaveValue('');
-    expect(boundaryLevels[1]).toHaveTextContent('Boundary level 1');
-    expect(boundaryLevels[1]).toHaveValue('1');
-    expect(boundaryLevels[2]).toHaveTextContent('Boundary level 2');
-    expect(boundaryLevels[2]).toHaveValue('2');
-    expect(boundaryLevels[3]).toHaveTextContent('Boundary level 3');
-    expect(boundaryLevels[3]).toHaveValue('3');
+    expect(screen.getByLabelText('Default boundary level')).not.toHaveValue();
+    const [defaultBoundaryLevelNullOption, ...defaultBoundaryLevelOptions] =
+      within(screen.getByLabelText('Default boundary level')).getAllByRole(
+        'option',
+      );
+
+    expect(defaultBoundaryLevelOptions).toHaveLength(
+      testMeta.boundaryLevels.length,
+    );
+
+    expect(defaultBoundaryLevelNullOption).toHaveTextContent('Please select');
+    expect(defaultBoundaryLevelNullOption).toHaveValue('');
+
+    defaultBoundaryLevelOptions.forEach((option, index) => {
+      const { id, label } = testMeta.boundaryLevels[index];
+      expect(option).toHaveTextContent(label);
+      expect(option).toHaveValue(String(id));
+    });
+
+    expect(
+      screen.getByText('Set boundary levels per data set'),
+    ).toBeInTheDocument();
+    expect(screen.queryByRole('table')).toBeInTheDocument();
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const [headerRow, ...dataRows] = screen.getAllByRole('row');
```

Comment:
Don't destructure like this. It's not good code hygiene to leave unused variables as we're unnecessarily assigning variables to memory for no reason, hence the lint error.

Seems like part of the problem is that we're still trying to be too smart with the for loop at the end. Similar to above comment, I wasn't suggest that we keep the for loop.

Would suggest just ditching the destructuring completely and doing something like this:

```ts
const rows = screen.getAllByRole('row');

expect(rows).toHaveLength(3);

const row1Cells = within(rows[1]).getAllByRole('cell');

expect(row1Cells[0]).toHaveTextContent(
  'Number of authorised absence sessions (Ethnicity Major Chinese, State-funded primary, All locations, 2014/15)',
);
expect(within(row1Cells[1]).getByRole('combobox')).toHaveValue('');

const row1BoundaryLevels = within(rows1Cells[1]).getAllByRole('option');
expect(row1BoundaryLevels[0]).toHaveTextContent('Use default');
expect(row1BoundaryLevels[0]).toHaveValue('');
expect(row1BoundaryLevels[1]).toHaveTextContent('Boundary level 1');
expect(row1BoundaryLevels[1]).toHaveValue('1');
expect(row1BoundaryLevels[2]).toHaveTextContent('Boundary level 2');
expect(row1BoundaryLevels[2]).toHaveValue('2');
expect(row1BoundaryLevels[3]).toHaveTextContent('Boundary level 3');
expect(row1BoundaryLevels[3]).toHaveValue('3');

const row2Cells = within(rows[2]).getAllByRole('cell');

// etc for row 2...
```

Again, this is a bit of copy and paste - but it's clear and precise.